geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java

```diff
@@ -265,6 +270,26 @@ public void removeInvalidXmlConfiguration(String xml, int expectedInitialElement
         .isEqualTo(expectFinalElements);
   }
 
+  @Test
+  public void dontUnlockSharedConfigurationIfNoLockedPreviously() {
+    InternalConfigurationPersistenceService service3;
+
+    MetricsService.Builder metricsSessionBuilder = mock(MetricsService.Builder.class);
+    when(metricsSessionBuilder.build(any())).thenReturn(mock(MetricsService.class));;
+
+    InternalDistributedSystem.Builder b =
+        new InternalDistributedSystem.Builder(new Properties(), metricsSessionBuilder);
+    InternalDistributedSystem ds = b.build();
+
+    service3 = spy(new InternalConfigurationPersistenceService(
+        DistributedLockService.create("test", ds),
+        JAXBService.createWithValidation(CacheConfig.class)));
+    doReturn(false).when(service3).lockSharedConfiguration();
+
+    service3.createConfigurationResponse(null);
```

Comment:
Without the code changes, this test fails in line 289. As Im mocking the call to `lockSharedConfiguration()` to return `false`, an exception is thrown when trying to unlock shared config inside `createConfigurationResponse`